internal/service/resiliencehub/app.go

```diff
@@ -0,0 +1,1502 @@
+				importInput := &resiliencehub.ImportResourcesToDraftAppVersionInput{
+					AppArn:           plan.ID.ValueStringPointer(),
+					SourceArns:       sourceArns,
+					TerraformSources: terraformSources,
+				}
+
+				_, err := conn.ImportResourcesToDraftAppVersion(ctx, importInput)
+				if err != nil {
+					resp.Diagnostics.AddError(
+						create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionUpdating, ResNameApp, plan.ID.ValueString(), err),
+						err.Error(),
+					)
+					return
+				}
+			}
+		}
+	}
+
+	// Handle tag updates - use TagsAll to include provider tags
+	if !plan.TagsAll.Equal(state.TagsAll) {
+		if err := updateTags(ctx, conn, plan.ID.ValueString(), state.TagsAll, plan.TagsAll); err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionUpdating, ResNameApp, plan.ID.ValueString(), err),
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	updateTimeout := r.UpdateTimeout(ctx, plan.Timeouts)
+	app, err := waitAppUpdated(ctx, conn, plan.ID.ValueString(), updateTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionWaitingForUpdate, ResNameApp, plan.ID.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	// Update computed values from the returned app
+	if app.DriftStatus != "" {
+		plan.DriftStatus = types.StringValue(string(app.DriftStatus))
+	} else {
+		plan.DriftStatus = types.StringValue("NotChecked")
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *appResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state appResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().ResilienceHubClient(ctx)
+
+	_, err := conn.DeleteApp(ctx, &resiliencehub.DeleteAppInput{
+		AppArn: state.ID.ValueStringPointer(),
+	})
+
+	if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		return
+	}
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionDeleting, ResNameApp, state.ID.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	deleteTimeout := r.DeleteTimeout(ctx, state.Timeouts)
+	_, err = waitAppDeleted(ctx, conn, state.ID.ValueString(), deleteTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionWaitingForDeletion, ResNameApp, state.ID.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func FindAppByARN(ctx context.Context, conn *resiliencehub.Client, arn string) (*awstypes.App, error) {
+	input := &resiliencehub.DescribeAppInput{
+		AppArn: aws.String(arn),
+	}
+
+	output, err := conn.DescribeApp(ctx, input)
+	if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		return nil, &retry.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
```

Comment:
Prefer `retry.NotFoundError` from `github.com/hashicorp/terraform-provider-aws/internal/retry`

```suggestion
		return nil, &retry.NotFoundError{
			LastError: err,
		}
```